streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -527,6 +529,54 @@ class StreamingContext private[streaming] (
     scheduler.listenerBus.addListener(streamingListener)
   }
 
+  /**
+   * Registers streamingListeners specified in spark.streaming.listeners
+   */
+  private def setupStreamingListeners(): Unit = {
+    // Use reflection to instantiate listeners specified via `spark.extraListeners`
+    try {
+      val listenerClassNames: Seq[String] =
+        conf.get("spark.streaming.listeners", "").split(',').map(_.trim).filter(_ != "")
+      for (className <- listenerClassNames) {
+        // Use reflection to find the right constructor
+        val constructors = {
+          val listenerClass = Class.forName(className)
```

Comment:
Please use `org.apache.spark.util.Utils.classForName` instead. `Class.forName` has problems with classloaders. I know this was copied verbatim from SparkContext. I verified @JoshRosen that it is incorrect in SparkContext. Mind updating that as well?